epsagon/events/httplib2.py

```diff
@@ -0,0 +1,146 @@
+"""
+httplib2 events module.
+"""
+
+from __future__ import absolute_import
+try:
+    from urllib.parse import urlparse
+except ImportError:
+    from urlparse import urlparse
+import traceback
+from uuid import uuid4
+import simplejson
+
+from epsagon.utils import add_data_if_needed
+from ..trace import tracer
+from ..event import BaseEvent
+from ..wrappers.http_filters import is_blacklisted_url
+from ..utils import update_api_gateway_headers
+
+
+class Httplib2Event(BaseEvent):
+    """
+    Represents base gttplib2 event.
+    """
+
+    ORIGIN = 'httplib2'
+    RESOURCE_TYPE = 'http'
+
+    #pylint: disable=W0613
+    def __init__(self, wrapped, instance, args, kwargs, start_time, response,
+                 exception):
+        """
+        Initialize.
+        :param wrapped: wrapt's wrapped
+        :param instance: wrapt's instance
+        :param args: wrapt's args
+        :param kwargs: wrapt's kwargs
+        :param start_time: Start timestamp (epoch)
+        :param response: response data
+        :param exception: Exception (if happened)
+        """
+
+        super(Httplib2Event, self).__init__(start_time)
+        self.event_id = 'httplib2-{}'.format(str(uuid4()))
+
+        # Params can be set via args or kwargs.
+        url = kwargs.get('uri', args[0] if len(args) >= 1 else 'N/A')
+        method = kwargs.get('method', args[1] if len(args) > 1 else 'N/A')
+        body = kwargs.get('body', args[2] if len(args) > 2 else None)
+        headers = kwargs.get('headers', args[3] if len(args) > 3 else None)
+
+        url_obj = urlparse(url)
+        self.resource['name'] = url_obj.hostname
+        self.resource['operation'] = method
+        self.resource['metadata']['url'] = url
+
+        if headers:
+            add_data_if_needed(
+                self.resource['metadata'],
+                'request_headers',
+                headers
+            )
+
+        try:
+            if body:
+                add_data_if_needed(
+                    self.resource['metadata'],
+                    'request_body',
+                    simplejson.loads(body)
+                )
+        except (TypeError, ValueError):
```

Comment:
There is a JSONDecodeError no?

Reply:
JSONDecodeError inherits from ValueError

Reply:
so catch the specific error... everything inherits from Object maybe just catch it and thats it